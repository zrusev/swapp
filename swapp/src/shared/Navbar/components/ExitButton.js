import styled from 'styled-components';
import logo from '../assets/logout.png';

export const ExitButton = styled.div`
    width: 25px;
    height: 25px;
    background-image: url(${logo});
    background-position: center;
    background-repeat: no-repeat;
`;