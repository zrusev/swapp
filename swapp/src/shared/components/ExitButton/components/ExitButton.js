import styled from 'styled-components';
import logo from '../assets/logout.png';

export const ExitButton = styled.div`
  width: 25px;
  height: 25px;
  background-image: url(${logo});
  background-position: center;
  background-repeat: no-repeat;
  padding: 0.8em 1em;

  &:hover,
  &:focus {
    box-shadow: rgba(255, 255, 255, 0.45) 0px 0px 10px 0px;
  }
`;
