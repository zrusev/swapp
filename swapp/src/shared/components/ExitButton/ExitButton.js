import React from 'react';
import PropTypes from 'prop-types';

import { ExitContainer, ExitButton as Button } from './components/_index';

const ExitButton = ({ action }) => (
  <ExitContainer onClick={action}>
    <span>
      <Button />
    </span>
  </ExitContainer>
);

ExitButton.propTypes = {
  action: PropTypes.func,
};

export default ExitButton;
