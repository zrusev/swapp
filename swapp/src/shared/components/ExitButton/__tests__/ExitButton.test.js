import React from 'react';
import { shallow, mount } from 'enzyme';

import ExitButton from '../ExitButton';

describe('exit button component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<ExitButton />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call passed function on click', () => {
    const props = {
      action: jest.fn(),
    };

    const wrapper = mount(<ExitButton {...props} />);

    wrapper.simulate('click');
    expect(props.action).toHaveBeenCalled();
  });
});
