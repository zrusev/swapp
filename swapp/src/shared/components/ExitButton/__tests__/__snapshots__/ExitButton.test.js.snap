// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exit button component should match snapshot 1`] = `
<styled.div>
  <span>
    <styled.div />
  </span>
</styled.div>
`;
